electron/main.mjs

```diff
@@ -79,6 +80,23 @@ function createWindow(url) {
     shell.openPath(arg);
   });
 
+  ipcMain.on('save-file', async (event, arg) => {
+    console.log('Saving file...');
+    try {
+      const result = await dialog.showSaveDialog({
+        defaultPath: path.join(app.getPath('downloads'), 'stacktrace.json'),
+      });
+      if (!result.canceled) {
+        fs.writeFileSync(result.filePath, arg);
+        console.log('File saved successfully.');
+      } else {
+        console.log('File save cancelled.');
+      }
+    } catch (err) {
```

Comment:
Not sure if it is easy enough to throw back the error to client and have client handle that